cypress/e2e/left-panel/left-panel.spec.ts

```diff
@@ -16,27 +16,746 @@ limitations under the License.
+
+                cy.get(".mx_LeftPanel_roomListWrapper").within(() => {
+                    // Assert the room tile is rendered
+                    cy.findByRole("treeitem", { name: roomName }).should("exist");
+                });
+            });
+        });
+    });
+
+    describe("for room list wrapper", () => {
+        let bot: MatrixClient;
+        const botName = "BotBob";
+        const message = "Message";
+
+        // Click "Room options" button which appears by hovering over the room tile
+        const clickRoomOptions = () => {
+            cy.get(".mx_LeftPanel_roomListWrapper").within(() => {
+                cy.findByRole("group", { name: "Rooms" }).within(() => {
+                    cy.findByRole("treeitem", { name: roomName })
+                        .realHover()
+                        .findByRole("button", { name: "Room options" })
+                        .click();
+                });
+            });
+        };
+
+        const createBotDM = () => {
+            // Create a bot
+            cy.getBot(homeserver, { displayName: botName }).then((_bot) => {
+                bot = _bot;
+            });
+
+            // Create DM with the bot
+            cy.getClient().then(async (cli) => {
+                const botRoom = await cli.createRoom({ is_direct: true });
+                await cli.invite(botRoom.room_id, bot.getUserId());
+                await cli.setAccountData("m.direct" as EventType, {
+                    [bot.getUserId()]: [botRoom.room_id],
+                });
+            });
+        };
+
+        const beforeSavedItems = () => {
+            // Enable Favorites
+            cy.setSettingValue("feature_favourite_messages", null, SettingLevel.DEVICE, true);
+        };
+
+        const beforeFavorites = () => {
+            cy.createRoom({ name: roomName });
+
+            clickRoomOptions();
+
+            // Click "Favorite" on the context menu
+            cy.findByRole("menuitemcheckbox", { name: "Favourite" }).click();
+
+            cy.get(".mx_LeftPanel_roomListWrapper").within(() => {
+                cy.findByRole("group", { name: "Favourites" }).should("exist");
+            });
+        };
+
+        const beforePeople = () => {
+            createBotDM();
+
+            cy.get(".mx_LeftPanel_roomListWrapper").within(() => {
+                cy.findByRole("group", { name: "People" }).should("exist");
+            });
+        };
+
+        const beforeHistorical = () => {
+            cy.createRoom({ name: roomName });
+
+            clickRoomOptions();
+
+            cy.findByRole("menuitem", { name: "Leave" }).click();
+            cy.findByRole("button", { name: "Leave" }).click();
+
+            cy.get(".mx_LeftPanel_roomListWrapper").within(() => {
+                cy.findByRole("group", { name: "Historical" }).should("exist");
+            });
+        };
+
+        const beforeLowPriority = () => {
```

Comment:
Please note these consts are reused by `should render sublists and take a Percy snapshot` below.